File: Backend/backend/src/main/java/com/example/backend/dto/user/UserDto.java
package com.example.backend.dto.user;

public class UserDto {
    private Long id;
    private String login;
    private String email;
    private boolean admin;

    public UserDto() {}

    public UserDto(Long id, String login, String email, boolean admin) {
        this.id = id;
        this.login = login;
        this.email = email;
        this.admin = admin;
    }

    public Long getId()              { return id; }
    public void setId(Long id)       { this.id = id; }

    public String getLogin()             { return login; }
    public void setLogin(String login)   { this.login = login; }

    public String getEmail()             { return email; }
    public void setEmail(String email)   { this.email = email; }

    public boolean isAdmin()             { return admin; }
    public void setAdmin(boolean admin)  { this.admin = admin; }
}

File: KotlinClient/app/src/main/java/com/example/kotlinclient/api_client/dto/UserDto.kt
package com.example.kotlinclient.api_client.dto

data class UserDto(
    val id: Long,
    val login: String,
    val email: String,
    val admin: Boolean
)

data class UpdateUserRequest(
    val login: String?,
    val email: String?
)

data class ChangePasswordRequest(
    val oldPassword: String,
    val newPassword: String
)
